src/fsharp/FSharp.Core/array.fs

```diff
@@ -1136,23 +1128,22 @@ namespace Microsoft.FSharp.Collections
             [<CompiledName("Partition")>]
             let partition predicate (array : 'T[]) =
                 checkNonNull "array" array
-                let inputLength = array.Length
-                let lastInputIndex = inputLength - 1
+                let inputLength = array.Length                
 
                 let isTrue = Array.zeroCreateUnchecked inputLength
                 Parallel.For(0, inputLength, 
                     fun i -> isTrue.[i] <- predicate array.[i]
                     ) |> ignore
                 
                 let mutable trueLength = 0
-                for i in 0 .. lastInputIndex do
+                for i in 0 .. isTrue.Length-1 do
```

Comment:
I was going by a pattern of referring to the first array to appear in the loop.  For instance
`indexed` uses res.Length-1 because res appears before array.  There is no logical reason for that, that was just my internal consistency .

Happy to change if if you like the array.Length-1 pattern better though